### ISSUE TITLE
Mecha Parts Uncraftable & Similar Items from other Mods Replacing Flans Items in Crafting Recipes

### ISSUE THREAD

**Issue by TwiliDiamondOcelot** [FlansMods/FlansMod#1092]
~~I have found while crafting mecha stuff that nearly all of the mecha items no longer have crafting recipes.~~ Also the Flotation Device uses the tin or copper canisters from galacticraft when installed instead of the flans canisters, and that the Caterpillar Track uses the small wheel from MrCrayfish's Device Mod when installed instead of the flans small wheel. ~~I don't know if these recipes work as normal when only flans is installed.~~

~~Anyways, here is the list of the mecha parts that don't have crafting recipes:~~
- ~~Rotary Magnet~~
- ~~Super Rotary Magnet~~
- ~~Redstone Magnet~~
- ~~Diamond Cutter~~
- ~~Laser Diamond Cutter~~
- ~~Metal Magpie~~
- ~~Ore Crusher 2000~~
- ~~Iron Drill~~
- ~~Iron Refiner~~
- ~~Auto Fueler~~
- ~~Waste Compacter~~
- ~~Iron Armor~~
- ~~Diamond Armor~~
- ~~Nanobots~~
- ~~Lantern~~
- ~~Illuminator~~
- ~~Diamond Detector (it doesn't even work so I don't care about it being uncraftable)~~
- ~~Rocket Pack~~

Forge Version: 14.23.5.2837
Flan's Mod & Content Pack Versions: 5.6

*Edit: struck through items have been adressed*

**Comment by TwiliDiamondOcelot** [comment]
Decided to look at the files and found the easiest fix! In all the mecha items that are uncraftable, the files don't have their names at the beginning of the file. For example, the first line of the Infinite ammo box reads "Name Infinite Ammo Box", while that of the Iron Drill doesn't have "Name Iron Drill". Add them and the items are craftable.

**Comment by TwiliDiamondOcelot** [comment]
Fix for the canister and the wheel recipe issues is to add a more specific identifier in their ShortName, ie flanwheel.

**Comment by TwiliDiamondOcelot** [comment]
One thing with changing the ShortName's is that it replaces the item textures with the null black and purple pattern with some weird tiny white letters/words.

**Comment by ChrisLane** [comment]
I've not checked but I suspect the real solution may be to add missing "name" values as in a17977170e2a90a4d1e11d7f5c4fda1c07e2fec8

**Comment by TwiliDiamondOcelot** [comment]
Yes that was the solution to the first problem, but the second problem I had in which I changed the ShortName's of the items was for mod compatibility. For example, another mod I have called "MrCrayfish's Furniture Mod" which also has a wheel item in it. The ShortName for the item is "wheel', which is the exact same ShortName as the Flans Mod small wheel. Now the Flans Mod small wheel is used in the crafting recipe for the Caterpillar Track, and the wheel from "MrCrayfish's Furniture Mod" overrides the wheel from Flans, and it is used in the recipe instead of the Flans wheel. Another instance is that the Flotation Device from the "Titan Pack" uses the canisters from "Galacticraft" instead of the canisters from the "Mecha Parts Pack" because they both have the ShortName canister. One thing I noticed is in all the mods is they specify that the item is from their mods, such as with galacticraft, "galacticraftcore:canister", which is absent in Flans. I also think that I should reopen this as it's still an issue.

**Comment by TwiliDiamondOcelot** [comment]
Found a fix. When changing the ShortName, I found out I needed to change naming in the assets/flansmod/lang in each lang file for the item to have the proper name. So if I made the ShortName for the small wheel "flansWheel", then I'd need to change the en-us.lang file "item.wheel.name=Small Wheel", to "item.flansWheel.name=Small Wheel". Doing both of these removes the compatibility issues with my other mods so I'll close this since there are solutions to both.
